src/libAtomVM/term.c

```diff
@@ -526,7 +534,7 @@ TermCompareResult term_compare(term t, term other, TermCompareOpts opts, GlobalC
                 break;
             }
             if (t_size > 0) {
-                for (int i = 1; i < t_size; i++) {
+                for (int i = t_size - 1; i >= 1; i--) {
```

Comment:
I am a little confused about this.  Is this change because of the ordering of keys?  Just out of curiosity, why wouldn’t we do the same thing with tuples?

Reply:
because tuples are wrong.